cumulusci/tasks/metadata_etl/layouts.py

```diff
@@ -68,6 +69,262 @@ def _create_new_related_list(self, metadata, api_name, related_list):
+            raise CumulusCIException(
+                f"Layout placement could not be determined for field: {new_item_name}"
+            )
+
+    def _process_position(self, position, default_position):
+        """Returns a MetadataElement that represents a  given the position"""
+        relative_position = position.get("relative", default_position.get("relative"))
+
+        # Mutually exclusive, therefore the higher priority is field if specified
+        relative_field_name = None
+        if "field" in position:
+            relative_field_name = (
+                self._inject_namespace(position.get("field"))
+                if position.get("field") is not None
+                else None
+            )
+            relative_section_index = None
+            if relative_position in ("top", "bottom"):
+                # Missing a relative position to the field, therefore default to after
+                relative_position = "after"
+        elif "section" in position:
+            relative_section_index = position.get("section")
+        else:
+            relative_section_index = default_position.get("section")
+            relative_position = default_position.get("position")
+
+        # Field relative
+        if relative_field_name is not None and relative_position in (
+            "before",
+            "after",
+        ):
+            new_layout_item = self._new_layout_item(
+                relative_field_name, relative_position
+            )
+
+        # Section relative
+        if relative_section_index is not None and relative_position in (
+            "top",
+            "bottom",
+        ):
+            new_layout_item = self._new_section_item(
+                relative_section_index,
+                relative_position,
+                position.get("column", default_position.get("column")),
+            )
+
+        return new_layout_item
+
+    def _new_layout_item(self, field_text, position):
+        for section in self._metadata.findall("layoutSections"):
+            for col in section.findall("layoutColumns"):
+                for item in col.findall("layoutItems"):
+                    for field in item.findall("field"):
+                        if field.text == field_text:
+                            return getattr(col, "insert_" + position)(
+                                item, "layoutItems"
+                            )
+
+    def _new_section_item(self, index, position, column_index):
+        # Get section at index
+        sections = [section for section in self._metadata.findall("layoutSections")]
+
+        if index > (len(sections) - 1):
+            self.logger.warning(
+                f"Unable to find section at index: {str(index)}. Default section is being selected."
+            )
+            return None
+
+        section = sections[index]
+        columns = [col for col in section.findall("layoutColumns")]
+        column = columns[1 if column_index == "last" else 0]
```

Comment:
If there's only one column, this could `IndexError`
```suggestion
        column = columns[-1 if column_index == "last" else 0]
```